extensions/walletProviders/kadena/src/extension.ts

```diff
@@ -0,0 +1,79 @@
+import { ITatumSdkContainer, Network, TatumConfig, TatumSdkWalletProvider } from '@tatumio/tatum'
+import { KadenaNetworkId, KadenaTxPayload, KadenaWallet } from './types'
+import { genKeyPair } from "@kadena/cryptography-utils/lib/genKeyPair";
+import { restoreKeyPairFromSecretKey } from "@kadena/cryptography-utils";
+import { ChainId, createClient, createSignWithKeypair, ICommand } from "@kadena/client";
+import { ICommandPayload } from "@kadena/types";
+import { KadenaLoadBalancerRpc } from "@tatumio/tatum/dist/src/service/rpc/other/KadenaLoadBalancerRpc";
+
+export class KadenaWalletProvider extends TatumSdkWalletProvider<KadenaWallet, KadenaTxPayload> {
+  private readonly kadenaRpc: KadenaLoadBalancerRpc
+  private readonly sdkConfig: TatumConfig
+
+  constructor(tatumSdkContainer: ITatumSdkContainer) {
+    super(tatumSdkContainer)
+    this.sdkConfig = this.tatumSdkContainer.getConfig()
+    this.kadenaRpc = this.tatumSdkContainer.get(KadenaLoadBalancerRpc)
+  }
+
+  /**
+   * Gets a Kadena wallet, which includes an address, secret key, and a public key.
+   * @returns {KadenaWallet} An object containing address, secret key, and public key.
+   */
+  public async getWallet() {
```

Comment:
Generally this looks good, but perhaps we should also include `account` creation to avoid confusion. As far as I understand, you still need to associate your public key with real `account` on kadena before you can use it as a real `wallet`, at least based on my own experiments, or am I wrong?